tpmutil/encoding_test.go

```diff
@@ -90,85 +92,71 @@ type nestedSlice struct {
 }
 
 func TestEncodingPackedSize(t *testing.T) {
-	if packedSize([]interface{}{uint32(3)}) != 4 {
-		t.Fatal("packedSize returned the wrong size for a uint32")
-	}
-
-	b := make([]byte, 10)
-	if packedSize([]interface{}{b}) != 14 {
-		t.Fatal("packedSize returned the wrong size for a byte slice")
-	}
-	if packedSize([]interface{}{&b}) != 14 {
-		t.Fatal("packedSize returned the wrong size for a pointer to a byte slice")
-	}
-
-	sp := simplePacked{137, 138}
-	if packedSize([]interface{}{sp}) != 8 {
-		t.Fatal("packedSize returned the wrong size for a simple struct")
-	}
-
-	np := nestedPacked{sp, 139}
-	if packedSize([]interface{}{np}) != 12 {
-		t.Fatal("packedSize returned the wrong size for a nested struct")
-	}
-
-	ns := nestedSlice{137, b}
-	if packedSize([]interface{}{ns}) != 18 {
-		t.Fatal("packedSize returned the wrong size for a struct that contains a slice")
-	}
-
-	// Pack an empty array; it should become uint32(0).
-	var empty []byte
-	if packedSize([]interface{}{empty}) != 4 {
-		t.Fatal("packType failed for an empty byte slice")
+	buf := make([]byte, 10)
+	tests := []struct {
+		in   interface{}
+		want int
+	}{
+		{uint32(3), 4},
+		{buf, 14},
+		{&buf, 14},
+		{simplePacked{137, 138}, 8},
+		{nestedPacked{simplePacked{137, 138}, 139}, 12},
+		{nestedSlice{137, buf}, 18},
+		{[]byte(nil), 4},
+	}
+	for _, tt := range tests {
+		if s, err := packedSize(tt.in); err != nil || s != tt.want {
+			t.Errorf("packedSize(%#v): %d, want %d", tt.in, s, tt.want)
+		}
 	}
 }
 
 func TestEncodingPackType(t *testing.T) {
```

Comment:
Can this test also be turned into a table-based test?

Reply:
Done